### DISCUSSION TITLE
Welcome to homeassistant-selve Discussions!

### DISCUSSION THREAD

**Discussion by Kannix2005** [Kannix2005/homeassistant-selve#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by OxYmOxY** [comment]
Hey,
thank you for your work.

I use the Selve USB-RF gateway via Selve HomeServer (first version) to control Commeo roller shutters, but wanted to transfer them to Homeassistant.

Connected the Selve usb-rf gateway with my device, which runs Homeassistant, installed "Selve NG" via HACS, but
nothing after that happens.
USB-RF gateway is recognized and available as /dev/ttyUSB0. But even with connected RF-Gateway and restart, "Selve NG" is not recognized as Integration in HA.

Any recommendations?

**Reply by Kannix2005** [reply]
Sorry for the late reply, i didn´t see the notification.
Hmm this sound weird. Did you figure it out?
If not, please have a look at the logs and post them here.
Nevertheless you can try to set the port to the device by id, eg. /dev/by-id/xxxx12345

**Reply by Kannix2005** [reply]
or maybe try to write this to your configuration.yaml:

`
selve:
  port: /dev/serial/by-id/usb-FTDI_FT230X_Basic_UART_DM01F76A-if00-port0
`

**Comment by frankfrommelt** [comment]
Hi Kannix2005,
I just added Selve NG to my Home assistant installation.
And it works like a charm! Thank you for your work!

I just added the integration via HACS and gave it the correct port.
I my case it is /dev/ttyUSB0, the "long" version /dev/serial/by-id/usb-FTDI_FT230X_Basic_UART_DM01F6GQ-if00-port0 did not work here.

The only thing that I consider a bit tricky is to copy the python libraries  https://github.com/Kannix2005/python-selve to the correct place.
The files should be placed i /config/python_scripts/selve.

Maybe you can add this step to the instrcutions/documentation.
In the configration flow the integration might check the presence of python-selve and could raise a specific error message if the libs are not available.

Thanks again

    Frank

**Reply by Kannix2005** [reply]
This is not the default behaviour. The library is automatically installed when the integration is added. So you shouldn´t need to download that. On which system did you try this?
I don ´t need to copy the Lib on Hassio. So maybe it´s some edge case.

Best
Stefan

**Comment by frankfrommelt** [comment]
Hi Stefan,

well, I did my first install of SELVE NG with on older version.

I tried to get past the “invalid port” message by installing the library.
But that did not change a thing.

Today I just gave it a new attempt wit the current version of SELVE NG and it worked.

No idea what chaged…

I am running a Home assistant OS.


Gruß

                      Frank

Von: Hebezo ***@***.***>
Gesendet: Donnerstag, 8. September 2022 16:38
An: Kannix2005/homeassistant-selve ***@***.***>
Cc: Frank Frommelt ***@***.***>; Comment ***@***.***>
Betreff: Re: [Kannix2005/homeassistant-selve] Welcome to homeassistant-selve Discussions! (Discussion #1)


This is not the default behaviour. The library is automatically installed when the integration is added. So you shouldn´t need to download that. On which system did you try this?
I don ´t need to copy the Lib on Hassio. So maybe it´s some edge case.

Best
Stefan
